test/ts_defines.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2021 Martin Jäger / Libre Solar
+ */
+
+#ifndef TS_DEFINES_H_
+#define TS_DEFINES_H_
+
+/*
+ * Categories / first layer node IDs
+ */
+#define ID_ROOT     0x00
+#define ID_INFO     0x18        // read-only device information (e.g. manufacturer, device ID)
+#define ID_CONF     0x30        // configurable settings
+#define ID_INPUT    0x60        // input data (e.g. set-points)
+#define ID_OUTPUT   0x70        // output data (e.g. measurement values)
+#define ID_REC      0xA0        // recorded data (history-dependent)
+#define ID_CAL      0xD0        // calibration
+#define ID_EXEC     0xE0        // function call
+#define ID_PUB      0xF0        // publication setup
+#define ID_SUB      0xF1        // subscription setup
+#define ID_LOG      0x100       // access log data
+
+#define PUB_SER     (1U << 0)   // UART serial
+#define PUB_CAN     (1U << 1)   // CAN bus
+#define PUB_NVM     (1U << 2)   // data that should be stored in EEPROM
+
+#endif /* TS_DEFINES_H_ */
```

Comment:
It is not clear whether these are only defines for the testsuite or general defines.

In case of the testsuite I think it is not worth to have an additional header. PR #10 uses test.h that also holds all test defines.

In case these are general defines they should be added to thingset.h. In this case the defines should also get a TS_ prefix to avoid clashing with other software.

Reply:
These defines are part of the specification as an example, but the user is free to choose the groups and IDs, so I would say it should be part of the test suite.

I removed the `ts_defines.h`, which was originally added to avoid duplicating this section in two files. Now the clean-up with `test.h` provides the single source of truth already.